theseus/core/cost_function.py

```diff
@@ -17,6 +17,19 @@
 from .variable import Variable
 
 
+def _register_vars_in_list(
```

Comment:
Trying to understand the underlying reason for the change of this from a class function to outside the class.

Reply:
We also use it in the `Vectorize` class. But now that I think about it we can probably make it a private function of base `TheseusFunction` class.

Reply:
Another option for such commonly used things might be to put them in utils.

Reply:
I replaced by a public `TheseusFunction.register_vars()`.